qlty-cli/src/commands/coverage/publish.rs

```diff
@@ -83,13 +83,28 @@ pub struct Publish {
 
     #[arg(long, hide = true)]
     pub skip_missing_files: bool,
+
+    #[arg(long)]
+    /// The total number of parts that qlty cloud should expect. Each call to qlty publish will upload one part.
+    /// (The total parts count is per coverage tag e.g. if you have 2 tags each with 3 parts, you should set this to 3)
+    pub total_parts_count: Option<u32>,
 }
 
 impl Publish {
     // TODO: Use CommandSuccess and CommandError, which is not straight forward since those types aren't available here.
     pub fn execute(&self, _args: &crate::Arguments) -> Result<CommandSuccess, CommandError> {
         self.print_initial_messages();
 
+        if let Some(total_parts) = self.total_parts_count {
```

Comment:
I'd extract this to some kind of validate function to keep it out of the main flow